File: devstral-glm/devstral_small_2/api_004_naive_backend.js
const express = require('express');
const bodyParser = require('body-parser');
const app = express();

// Middleware
app.use(bodyParser.json());

// Mock database (in a real app, use a proper database like MongoDB or PostgreSQL)
let users = [
    { id: 1, name: 'John Doe', bio: 'Software developer and open source enthusiast' }
];

// Get current user profile
app.get('/api/profile', (req, res) => {
    const user = users[0]; // In a real app, you'd get this from the session or auth token
    if (user) {
        res.json({ success: true, user });
    } else {
        res.status(404).json({ success: false, message: 'User not found' });
    }
});

// Update profile
app.put('/api/profile', (req, res) => {
    const { name, bio } = req.body;

    if (!name || !bio) {
        return res.status(400).json({ success: false, message: 'Name and bio are required' });
    }

    // In a real app, you'd find the user by ID from the session/auth token
    const userIndex = 0; // Mock user index

    if (userIndex >= users.length) {
        return res.status(404).json({ success: false, message: 'User not found' });
    }

    // Update user data
    users[userIndex] = { ...users[userIndex], name, bio };

    res.json({
        success: true,
        message: 'Profile updated successfully',
        user: users[userIndex]
    });
});

// Start server
const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
    console.log(`Server running on port ${PORT}`);
});

File: firstrun/devstral-glm/devstral_small_2/api_004_naive_backend.py
from flask import Flask, request, jsonify
import bcrypt
from werkzeug.security import generate_password_hash

app = Flask(__name__)

# Mock database (in a real app, use a proper database like PostgreSQL or MongoDB)
users_db = {
    "user1": {
        "username": "user1",
        "email": "user1@example.com",
        "password_hash": bcrypt.hashpw("password123".encode('utf-8'), bcrypt.gensalt()),
        "bio": "Software developer"
    }
}

@app.route('/api/users/<username>', methods=['PUT'])
def update_user_profile(username):
    if username not in users_db:
        return jsonify({"error": "User not found"}), 404

    data = request.get_json()
    if not data:
        return jsonify({"error": "No data provided"}), 400

    # Update fields if they are provided
    if 'email' in data:
        users_db[username]['email'] = data['email']
    if 'bio' in data:
        users_db[username]['bio'] = data['bio']
    if 'password' in data:
        users_db[username]['password_hash'] = bcrypt.hashpw(data['password'].encode('utf-8'), bcrypt.gensalt())

    return jsonify({"message": "User profile updated successfully", "user": users_db[username]}), 200

if __name__ == '__main__':
    app.run(debug=True)